models/user.rb

```diff
@@ -232,14 +233,32 @@ def replace_username_in_all_content(new_username)
     end
   end
 
-
   def mark_as_read(thread)
     reconnect_mongo_primary
     read_state = read_states.find_or_create_by(course_id: thread.course_id)
     read_state.last_read_times[thread.id.to_s] = Time.now.utc
     read_state.save
   end
 
+  ##
+  # Update stats for a the user for the specified course by the specified amounts.
+  # The amounts here are increments/decrements, and not absolute
+  def update_stats_for_course(
+    course_id, threads: nil, responses: nil, replies: nil, active_flags: nil, inactive_flags: nil
+  )
+    begin
+      stats = course_stats.find(course_id: course_id)
+      stats.inc(threads: threads) if threads
+      stats.inc(responses: responses) if responses
+      stats.inc(replies: replies) if replies
+      stats.inc(active_flags: active_flags) if active_flags
+      stats.inc(inactive_flags: inactive_flags) if inactive_flags
```

Comment:
This is doing multiple roundtrips to mongodb every time multiple fields need to be updated.
Instead of defaulting to `nil`, this could simply default to `0` and pass all the parameters into a single `inc` call.

Reply:
@felipetrz It won't since we never increment more than one item at a time. At any point only one of these parameters will be passed, so only one of these values will be incremented. 

That said, while I did this to keep the function explicit, I'll just simplify it so it's easier to follow.